### ISSUE TITLE
Some check in closeLedger are duplicated in applyHistoryOfSingleLedger

### ISSUE THREAD

**Issue by vogel** [stellar/stellar-core#1638]
applyHistoryOfSingleLedger method duplicates some check from closeLedger. We don't need these in both places - closeLedger is enough.

Compare:
```
if (header.previousLedgerHash != lm.getLastClosedLedgerHeader().hash)
if (ledgerData.getTxSet()->previousLedgerHash() !=
        getLastClosedLedgerHeader().hash)

if (ledgerData.getTxSet()->getContentsHash() !=
        ledgerData.getValue().txSetHash)
if (header.scpValue.txSetHash != txset->getContentsHash())
```

**Comment by MonsieurNicolas** [comment]
no, we need to perform the check in both places: one is validating the ledger chain (only using ledger headers, before downloading buckets etc), the other is applying a specific txset that comes from consensus.

If you're saying that we need some abstraction to compare hashes, we already do: it's the `operator==` on `Hash`.

**Comment by MonsieurNicolas** [comment]
Oh I see - yes, `applyHistoryOfSingleLedger` should not check much.

I think that it should just:
* check the number (before -> skip, otherwise fallback to call `closeLedger`)
* handle `closeLedger` throwing

In addition, we should probably add a check in `closeLedger` (at the top) for the ledger sequence number (right now only checks the hash - which is too cryptic to debug if the ledger number was wrong)
